tests/test_general_linear.py

```diff
@@ -9,105 +9,56 @@
 import geomstats.backend as gs
 import geomstats.tests
 from geomstats.geometry.general_linear import GeneralLinear
-from geomstats.geometry.special_orthogonal import SpecialOrthogonal
 
 RTOL = 1e-5
 
-
 class TestGeneralLinearMethods(geomstats.tests.TestCase):
     def setUp(self):
         gs.random.seed(1234)
         self.n = 3
         self.n_samples = 2
         self.group = GeneralLinear(n=self.n)
-        # We generate invertible matrices using so3_group
-        self.so3_group = SpecialOrthogonal(n=self.n)
 
         warnings.simplefilter('ignore', category=ImportWarning)
 
     @geomstats.tests.np_only
     def test_belongs(self):
-        """
-        A rotation matrix belongs to the matrix Lie group
-        of invertible matrices.
-        """
-        rot_vec = gs.array([0.2, -0.1, 0.1])
-        rot_mat = self.so3_group.matrix_from_rotation_vector(rot_vec)
-        result = self.group.belongs(rot_mat)
-        expected = gs.array([[True]])
-
+        mats = gs.array([
+            [[1., 0.], [0., 1]],
+            [[0., 1.], [0., 1.]]])
+        result = self.group.belongs(mats)
+        expected = gs.array([True, False])
         self.assertAllClose(result, expected)
 
-    def test_compose(self):
-        # 1. Composition by identity, on the right
-        # Expect the original transformation
-        rot_vec = gs.array([0.2, -0.1, 0.1])
-        mat = self.so3_group.matrix_from_rotation_vector(rot_vec)
-
-        result = self.group.compose(mat, self.group.identity)
-        expected = mat
-        expected = helper.to_matrix(mat)
-
+    def test_compose_and_identity(self):
+        mat1 = gs.array([
+            [1., 0.],
+            [0., 2.]])
+        mat2 = gs.array([
+            [2., 0.],
+            [0., 1.]])
+        result = self.group.compose(mat1, mat2)
+        expected = 2. * GeneralLinear(2).identity()
         self.assertAllClose(result, expected)
 
-        # 2. Composition by identity, on the left
-        # Expect the original transformation
-        rot_vec = gs.array([0.2, 0.1, -0.1])
-        mat = self.so3_group.matrix_from_rotation_vector(rot_vec)
-
-        result = self.group.compose(self.group.identity, mat)
-        expected = mat
-
-        self.assertAllClose(result, expected)
-
-    def test_inverse(self):
+    def test_inv(self):
         mat = gs.array([
             [1., 2., 3.],
             [4., 5., 6.],
             [7., 8., 10.]])
-        result = self.group.inverse(mat)
+        result = self.group.inv(mat)
         expected = 1. / 3. * gs.array([
             [-2., -4., 3.],
             [-2., 11., -6.],
             [3., -6., 3.]])
-        expected = helper.to_matrix(expected)
-
-        self.assertAllClose(result, expected)
-
-    def test_compose_and_inverse(self):
```

Comment:
You removed this in order to have unit tests that are really "unit"?

Reply:
yes there were many tests to fix so I removed those that seemed redundant!

Reply:
I might have been a bit quick doing so, so let me know if you think something's missing...
I should ask one for `orbit` BTW